### PR TITLE
Documented the new Inflector component

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | master (3.1+) |
| Fixed tickets | #6417 |

Excuse me if this question is stupid, but while documenting this component I wondered why there isn't a reverse `pluralize()` method.

### CODE REVIEW

**Comment by xabbuh** [comment]
@javiereguiluz Because the Form component didn't need it. :)

File: components/inflector.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+.. index::
+   single: Inflector
+   single: Components; Inflector
+
+The Inflector Component
+=======================
+
+    The Inflector component converts English words between singular and plural forms.
```
Comment (xabbuh) [inline components/inflector.rst]: "[...] from plural to singular."

File: components/inflector.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+* Use the official Git repository (https://github.com/symfony/inflector).
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+The Inflector component provides an easy and consistent way to convert English
+words between singular and plural forms. The static ``singularize()`` method
+defined by :class:`Symfony\\Component\\Inflector\\Inflector` class returns the
```
Comment (xabbuh) [inline components/inflector.rst]: [...] defined by the [...]

File: components/inflector.rst (hunk 1 above)
Comment (HeahDude) [inline components/inflector.rst]: In french, when using adjectives or the name of the language, we use lower case, upper case is for inhabitant. Is the rule different in "english" ? :)

Comment (xabbuh) [inline components/inflector.rst]: Afaik it's always capitalised in the English language: http://www.oxfordlearnersdictionaries.com/definition/english/english_2

Comment (HeahDude) [inline components/inflector.rst]: Thanks for that :)

**Comment by javiereguiluz** [comment]
I've fixed the reported issues.

@xabbuh thanks for the info. Anyway, my feel is that there's something wrong with this "component". Comparing it with the [Doctrine Inflector](https://github.com/doctrine/inflector/blob/master/lib/Doctrine/Common/Inflector/Inflector.php) this looks "useless".

**Comment by weaverryan** [comment]
@javiereguiluz, after checking into the code PR (because I agreed with your point about the Doctrine Inflector), I found that this component is marked internal, explicitly because we *may* actually break BC or even remove it in the future: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Inflector/README.md#disclaimer. So, I think we should *not* actually document this.

**Comment by javiereguiluz** [comment]
@weaverryan thanks for letting me know 😱  !  I was going to publish a "New in Symfony 3.1" about it too. Closing it.

**Pull request closed without merging**